src/openrct2/entity/Guest.cpp

```diff
@@ -7318,10 +7318,10 @@ bool Guest::UpdateQueuePosition(PeepActionType previous_action)
                 return false;
         }
 
-        if (sprite_direction != guestNext->sprite_direction)
+        if (PeepDirection != guestNext->PeepDirection)
             return false;
 
-        switch (guestNext->sprite_direction / 8)
+        switch (guestNext->PeepDirection)
```

Comment:
Could you explain why the `/ 8` was dropped?

Reply:
I'm not too sure about this, but I have see in other parts of the code that `sprite_direction >> 3` is used to set/calculate `PeepDirection`, and `PeepDirection << 3` to set `sprite_direction`. If I am not wrong, `sprite_direction / 8` is equivalent to `sprite_direction >> 3`. I do not know much about the logic of `sprite_direction` but I have see that a walking guest can have a `sprite_direction` of `0`, `8`, `16` and `24` (depending of the direction that guest is looking), and divide it by 8 will result in `0`, `1`, `2` and `3`, comparing this with `PeepDirection` I see that both have the same value:

![image](https://user-images.githubusercontent.com/7508197/212835975-6eeb07fa-9288-4683-8550-f7bc19cc21d4.png)

The reason of why I no use `sprite_direction / 8` here is due to the issue that I explained before: a queuing guest can wait in the queue looking at another direction different from the real direction that guest is currently following, and because of that, `sprite_direction` may not always reflect the real direction of a queuing guest. In other hand, `PeepDirection` always have the real direction of a queuing guest.

----
I have made some tests to test the `switch` in all directions.

The purpose of the `switch` is to allow a guest to walk along the queue if the guest is before to the next guest.

To trigger the `switch`, both guests must have a distance greater than 7 units from any axis (X,Y), if distance from both guests is also greater than 13 units then both guest must be on the same tile. Also both guests must have the same direction. 

I have aligned guests to the side of the queue instead of to the middle to accomplish with this condition.

* Test  `#1` using `switch(guestNext->sprite_direction / 8)`:
![test_sprite_direction](https://user-images.githubusercontent.com/7508197/212822500-c6207818-0156-42d7-b2b5-b233b7350795.gif)

* Test  `#2` using `switch(guestNext->PeepDirection)` (same result as Test `#1`):
![test_peep_direction](https://user-images.githubusercontent.com/7508197/212822522-7538a575-07db-437f-a90c-9a9c0c704b34.gif)

    also works for non-straight queue paths:
    ![test_no_straight_path](https://user-images.githubusercontent.com/7508197/212831383-ec732e13-d568-4e50-855f-795426f606c4.gif)

* Test  `#3` simulating that the switch is not implemented correctly:
![test_switch_fail](https://user-images.githubusercontent.com/7508197/212822567-eac8ba7f-f4ce-40c1-9dea-fc64ad59f740.gif)

I have inverted all cases of the switch to do this test:
```c++
switch(...)
{
   case 3:
      ...
      break;
   case 2:
      ...
      break;
   case 1:
      ...
      break;
   case 0:
      ...
      break;
}
```

* Test  `#4` simulating that the switch is not implemented correctly and none of the switch cases are reached:
![test_switch_no_trigger](https://user-images.githubusercontent.com/7508197/212823104-2898732d-10e8-4d39-bad3-ad6679b7d178.gif)

Reply:
PeepDirection is the direction that the peep is travelling on the tile. sprite_direction is used for the actual sprite that is drawn. sprite_direction has much higher granularity as it is also used for vehicles but for peeps it is always just * 8, <<3 of the cardinal directions. So this change does make sense I think.